pkg/hubble/exporter/aggregator.go

```diff
@@ -0,0 +1,199 @@
+type AggregateValue struct {
+	IngressFlowCount          int
+	EgressFlowCount           int
+	UnknownDirectionFlowCount int
+	ProcessedFlow             *flowpb.Flow
+}
+
+type Aggregator struct {
+	m              map[AggregateKey]*AggregateValue
+	mu             lock.RWMutex
+	fieldAggregate fa.FieldAggregate
+	logger         *slog.Logger
+}
+
+// NewAggregator creates a new aggregator without field aggregation
+func NewAggregator(logger *slog.Logger) *Aggregator {
+	return &Aggregator{
+		m:      make(map[AggregateKey]*AggregateValue),
+		logger: logger.With(logfields.LogSubsys, "hubble-aggregator"),
+	}
+}
+
+// NewAggregatorWithFields creates a new aggregator with field aggregation
+func NewAggregatorWithFields(fieldAggregate fa.FieldAggregate, logger *slog.Logger) *Aggregator {
+	return &Aggregator{
+		m:              make(map[AggregateKey]*AggregateValue),
+		fieldAggregate: fieldAggregate,
+		logger:         logger.With(logfields.LogSubsys, "hubble-aggregator"),
+	}
+}
+
+func (a *Aggregator) Add(ev *v1.Event) {
+	f := ev.GetFlow()
+	if f == nil {
+		return
+	}
+
+	processedFlow := &flowpb.Flow{}
+	a.fieldAggregate.Copy(processedFlow.ProtoReflect(), f.ProtoReflect())
+
+	k := a.generateAggregationKey(processedFlow)
+
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	v, ok := a.m[k]
+	if !ok {
+		switch f.GetTrafficDirection() {
+		case flowpb.TrafficDirection_INGRESS:
+			v = &AggregateValue{
+				IngressFlowCount: 1,
+				ProcessedFlow:    processedFlow,
+			}
+		case flowpb.TrafficDirection_EGRESS:
+			v = &AggregateValue{
+				EgressFlowCount: 1,
+				ProcessedFlow:   processedFlow,
+			}
+		default:
+			v = &AggregateValue{
+				UnknownDirectionFlowCount: 1,
+				ProcessedFlow:             processedFlow,
+			}
+		}
+		a.m[k] = v
+	} else {
+		switch f.GetTrafficDirection() {
+		case flowpb.TrafficDirection_INGRESS:
+			v.IngressFlowCount++
+		case flowpb.TrafficDirection_EGRESS:
+			v.EgressFlowCount++
+		default:
+			v.UnknownDirectionFlowCount++
+		}
+	}
+}
+
+func (a *Aggregator) generateAggregationKey(processedFlow *flowpb.Flow) AggregateKey {
+	b, _ := proto.Marshal(processedFlow.ProtoReflect().Interface())
+	return AggregateKey(b)
+}
+
+// AggregatorRunner manages the lifecycle of an aggregator with periodic export
+type AggregatorRunner struct {
+	aggregator *Aggregator
+	interval   time.Duration
+	encoder    Encoder
+	logger     *slog.Logger
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+func (r *AggregatorRunner) Start(ctx context.Context) {
+	r.ctx, r.cancel = context.WithCancel(ctx)
+	r.wg.Add(1)
+	go r.run()
+}
+
+func (r *AggregatorRunner) Stop() {
+	if r.cancel != nil {
+		r.cancel()
+		r.cancel = nil
+	}
+	r.wg.Wait()
+}
+
+func (r *AggregatorRunner) Add(event *v1.Event) {
+	r.aggregator.Add(event)
+}
+
+func (r *AggregatorRunner) run() {
+	defer r.wg.Done()
+	ticker := time.NewTicker(r.interval)
+	defer ticker.Stop()
```

Comment:
suggestion: since Go 1.23, it is no longer necessary to call Stop on a Ticker: https://pkg.go.dev/time#NewTicker